src/svcutilcore/src/Microsoft/Tools/ServiceModel/SvcUtil/XmlSerializerGenerator.cs

```diff
@@ -124,21 +122,94 @@ private List<Type> CollectXmlSerializerTypes(Assembly assembly, List<XmlMapping>
                         ToolConsole.WriteWarning(SR.Format(SR.WrnUnableToLoadContractForSGen, contractType, errorMessage));
                     };
 
-            foreach (ContractDescription contract in contractLoader.GetContracts())
+            Type contractDescriptionType = Tool.SMAssembly.GetType("System.ServiceModel.Description.ContractDescription");
+            if (contractDescriptionType == null)
             {
-                types.Add(contract.ContractType);
-                foreach (OperationDescription operation in contract.Operations)
+                ToolConsole.WriteError($"Not found type System.ServiceModel.Description.ContractDescription in {Tool.SMAssembly.FullName}");
+                throw new ToolRuntimeException();
+            }
+
+            foreach (object contract in contractLoader.GetContracts())
+            {
+                PropertyInfo contractTypeProperty = contractDescriptionType.GetProperty("ContractType");
+                if (contractTypeProperty == null)
+                {
+                    ToolConsole.WriteError($"Not found property ContractType in type {contractDescriptionType}");
+                    throw new ToolRuntimeException();
+                }
+
+                types.Add((Type)contractTypeProperty.GetValue(contract));
+                PropertyInfo operationsProperty = contractDescriptionType.GetProperty("Operations");
+                if (operationsProperty == null)
+                {
+                    ToolConsole.WriteError($"Not found property Operations in type {contractDescriptionType}");
+                    throw new ToolRuntimeException();
+                }
+
+                var operations = operationsProperty.GetValue(contract);
+                Type operationsType = operations.GetType();
+                MethodInfo getCount = operationsType.GetMethod("get_Count");
+                int count = (int)getCount.Invoke(operations, new object[] { });
+                for (int i = 0; i < count; ++i)
                 {
-                    XmlSerializerOperationBehavior behavior = operation.Behaviors.Find<XmlSerializerOperationBehavior>();
+                    MethodInfo getItem = operationsType.GetMethod("get_Item");
+                    var operation = getItem.Invoke(operations, new object[] { i });
```

Comment:
Move outside of foreach loop and change to getting the property "Item".

Reply:
And I cannot move it out of the loop, It reply on the operationsType, which get from operations